src/engine/source/api/src/test/handlers.cpp

```diff
@@ -0,0 +1,412 @@
+#include "api/test/handlers.hpp"
+
+#include <api/adapter.hpp>
+#include <api/catalog/resource.hpp>
+
+#include <eMessages/eMessage.h>
+#include <eMessages/test.pb.h>
+
+#include "api/test/sessionManager.hpp"
+
+namespace
+{
+
+using namespace api::sessionManager;
+
+using std::shared_ptr;
+using std::string;
+
+using ::api::adapter::fromWazuhRequest;
+using ::api::adapter::genericError;
+using ::api::adapter::toWazuhResponse;
+
+} // namespace
+
+namespace api::test::handlers
+{
+
+namespace eEngine = ::com::wazuh::api::engine;
+namespace eTest = ::com::wazuh::api::engine::test;
+
+/**
+ * @brief Get the minimum available priority for a route.
+ *
+ * @param router Router instance.
+ * @return int32_t Minimum available priority. If no priority is available, returns -1.
+ */
+static inline int32_t getMinimumAvailablePriority(const shared_ptr<::router::Router>& router)
+{
+    // Create a set to store the taken priorities given the table
+    std::unordered_set<uint32_t> takenPriorities;
+
+    const auto routerTable = router->getRouteTable();
+    for (const auto& [name, priority, filter, policy] : routerTable)
+    {
+        takenPriorities.insert(priority);
+    }
+
+    int32_t minAvailablePriority {MINIMUM_PRIORITY};
+
+    // The condition may be confusing as the actual priority increases while the value decreases
+    while (takenPriorities.count(minAvailablePriority) > 0 && MAXIMUM_PRIORITY > minAvailablePriority)
+    {
+        // If priority is taken, decrease the value (so, increase the priority)
+        minAvailablePriority--;
+    }
+
```

Comment:
**thought:** In `/src/engine/source/router/src/router.cpp` we can see at line 209 that `std::vector<Router::Entry> Router::getRouteTable()` returns a sorted table by Priority. Wouldn't this mean that despite the confusing order of priorities we can get the max or min priority to get the minAvailablePriority avoiding this loop?

Reply:
Priorities are not assigned in a continuous way, so it might be tricky actually to find the minimum available priority in that way.